packages/filesystem/src/node/node-filesystem.ts

```diff
@@ -94,15 +97,16 @@ export class FileSystemNode implements FileSystem {
             throw this.createOutOfSyncError(file, stat);
         }
         const encoding = await this.doGetEncoding(options);
-        await fs.writeFile(FileUri.fsPath(_uri), content, { encoding });
```

Comment:
What is better about `iconv` compare to passing `encoding` to node.js?

Reply:
Iconv-lite is better because its list of supported encoding is much longer than node.js. Encodings like GB2312 and Shift-JIS, which are shown in my demos, will not be supported in if we don't use a third-party package. It is a necessary approach to introduce a wide range of supported encodings if we want theia to be used worldwide. In addition to that, iconv-lite is a popular, if not the most popular package concerning encoding support, and is adopted by VSCode as well.

Reference
1. https://stackoverflow.com/questions/14551608/list-of-encodings-that-node-js-supports
2. https://github.com/ashtuchkin/iconv-lite/wiki/Supported-Encodings
3. https://github.com/microsoft/vscode/blob/master/src/vs/base/node/encoding.ts

Reply:
> What is better about `iconv` compare to passing `encoding` to node.js?

Iconv-lite is better because its list of supported encoding is much longer than node.js. Encodings like GB2312 and Shift-JIS, which are shown in my demos, will not be supported in if we don't use a third-party package. It is a necessary approach to introduce a wide range of supported encodings if we want theia to be used worldwide. In addition to that, iconv-lite is a popular, if not the most popular package concerning encoding support, and is adopted by VSCode as well.

Reference
1. https://stackoverflow.com/questions/14551608/list-of-encodings-that-node-js-supports
2. https://github.com/ashtuchkin/iconv-lite/wiki/Supported-Encodings
3. https://github.com/microsoft/vscode/blob/master/src/vs/base/node/encoding.ts

Reply:
Maybe we can add it as a comment to `iconv-lite` import int this file. Thank you for links.